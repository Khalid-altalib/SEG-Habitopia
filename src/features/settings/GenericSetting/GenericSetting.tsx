import { Button, Input, Text, TextArea, View } from "native-base";
import React from "react";
import { Controller, useForm } from "react-hook-form";
import { useDispatch, useSelector } from "../../../app/hooks";
import DoneButton from "../DoneButton/DoneButton";
import settingEntries from "../settingEntries";

type Props = {
  defaultValue: string;
  settingType: string;
};

const GenericSetting = (props: Props) => {
  const { defaultValue, settingType } = props;

  const { control, getValues } = useForm({
    defaultValues: {
      formValue: defaultValue,
    },
  });

  const settingTitle = settingEntries[settingType].name;

  return (
    <View>
      <Text mb={5}>{settingTitle}</Text>

      <Controller
        control={control}
        render={({ field: { onChange, value } }) => (
          <Input
            onChangeText={onChange}
            value={value}
            size="xl"
            testID="input"
          />
        )}
        name="formValue"
      />
      <DoneButton getValues={getValues} valueName={settingType} />
    </View>
  );
};

export default GenericSetting;
