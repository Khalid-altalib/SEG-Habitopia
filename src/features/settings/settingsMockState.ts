import { SettingsState } from "./settingsSlice";

/**
 * The mock state for the settings redux state
 */

const settingsMockState: SettingsState = {
  settings: {
    email: "test@test.com",
    password: "password",
    notifications: true,
    name: "Test User",
    biography: "Hello, I'm Test User.",
  },
  fetchSettings: {
    loading: false,
    error: "",
  },
  setSettings: {
    loading: false,
    error: "",
  },
};

export default settingsMockState;
