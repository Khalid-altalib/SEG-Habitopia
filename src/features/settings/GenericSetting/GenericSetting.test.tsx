import { render } from "@testing-library/react-native";
import configureStore from "redux-mock-store";
import thunk from "redux-thunk";
import TestingWrapper from "../../../app/TestingWrapper";
import settingsMockState from "../settingsMockState";
import GenericSetting from "./GenericSetting";

describe("GenericSetting", () => {
  const mockState = { settings: settingsMockState };
  const mockStore = configureStore([thunk])(mockState);

  const defaultProps = {
    defaultValue: "example@email.com",
    settingType: "email",
  };

  it("renders correctly", () => {
    const tree = render(
      <TestingWrapper store={mockStore}>
        <GenericSetting {...defaultProps} />
      </TestingWrapper>
    );

    expect(tree).toMatchSnapshot();
  });

  it("finds the input", () => {
    const tree = render(
      <TestingWrapper store={mockStore}>
        <GenericSetting {...defaultProps} />
      </TestingWrapper>
    );

    const input = tree.getByTestId("input");

    expect(input).toBeDefined();
  });
});
