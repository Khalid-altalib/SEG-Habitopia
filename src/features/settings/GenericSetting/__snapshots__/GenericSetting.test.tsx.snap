// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GenericSetting component should render correctly 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    dataSet={{}}
    style={{}}
  >
    <Text
      style={
        [
          {
            "color": "#18181b",
          },
          {
            "fontFamily": undefined,
            "fontSize": 19,
          },
          {
            "marginBottom": 10,
          },
        ]
      }
    >
      Name
    </Text>
    <View
      dataSet={{}}
      isFocused={false}
      style={
        {
          "alignItems": "center",
          "borderColor": "#d4d4d4",
          "borderRadius": 4,
          "borderWidth": 1,
          "flexDirection": "row",
          "overflow": "hidden",
        }
      }
    >
      <TextInput
        accessible={true}
        dataSet={{}}
        editable={true}
        onBlur={[Function]}
        onChangeText={[Function]}
        onFocus={[Function]}
        onKeyPress={[Function]}
        placeholderTextColor="#a3a3a3"
        secureTextEntry={false}
        style={
          {
            "backgroundColor": "transparent",
            "color": "#171717",
            "flex": 1,
            "fontFamily": undefined,
            "fontSize": 18,
            "fontStyle": "normal",
            "fontWeight": "400",
            "height": "100%",
            "paddingBottom": 8,
            "paddingLeft": 12,
            "paddingRight": 12,
            "paddingTop": 8,
            "width": "100%",
          }
        }
        testID="input"
        value="John Doe"
      />
    </View>
    <View
      dataSet={{}}
      style={{}}
    >
      <View
        accessibilityRole="button"
        accessible={true}
        collapsable={false}
        colorScheme="primary"
        dataSet={{}}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#0891b2",
            "borderRadius": 4,
            "flexDirection": "row",
            "justifyContent": "center",
            "marginTop": 32,
            "paddingBottom": 10,
            "paddingLeft": 12,
            "paddingRight": 12,
            "paddingTop": 10,
          }
        }
        testID="button"
      >
        <View
          dataSet={{}}
          style={
            {
              "alignItems": "center",
              "flexDirection": "row",
            }
          }
          test={true}
        >
          <View
            dataSet={{}}
            style={{}}
          >
            <Text
              dataSet={{}}
              style={
                {
                  "backgroundColor": undefined,
                  "color": "#fafafa",
                  "fontFamily": undefined,
                  "fontSize": 14,
                  "fontStyle": "normal",
                  "fontWeight": "400",
                  "letterSpacing": 0,
                  "lineHeight": 21,
                  "textDecorationLine": undefined,
                }
              }
            >
              Done
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</RNCSafeAreaProvider>
`;
